packages/lesswrong/lib/collections/dialogueChecks/collection.ts

```diff
@@ -0,0 +1,36 @@
+import {ensureIndex} from "../../collectionIndexUtils";
+import {addUniversalFields, getDefaultMutations, getDefaultResolvers, schemaDefaultValue} from "../../collectionUtils";
+import {createCollection} from "../../vulcan-lib";
+import schema from "./schema";
+
+export const DialogueChecks: DialogueChecksCollection = createCollection({
+  collectionName: 'DialogueChecks',
+  typeName: 'DialogueCheck',
+  collectionType: 'pg',
+  schema,
+  resolvers: getDefaultResolvers('DialogueChecks'),
+  logChanges: true,
+})
+
+DialogueChecks.checkAccess = async (user: DbUser|null, document: DbDialogueCheck, context: ResolverContext|null): Promise<boolean> => {
+  // Case 1: A user can see their own checks
+  if (document.userId === user?._id) {
+    return true;
+  }
+
+  // Case 2: A user can see the checks of people they themselves have checked... 
+  const existingCheck = await DialogueChecks.findOne({ userId: user?._id, targetUserId: document.userId, checked: true });
+  // ...but only the checks concerning themselves
+  const selfConcerningCheck = (document.targetUserId === user?._id)
+  if (existingCheck && selfConcerningCheck) {
```

Comment:
I was about to suggest doing an early return based on just the value of `selfConcerningCheck` (since if it's false you technically don't need to do the above db lookup), but in theory that might expose us to timing attacks, so... nevermind.